### PR TITLE
Merge front-end with back-end and added error handling

### PR DESCRIPTION
**Pull request by B-Tanonaka**
1. Closing previous PR about Profile page as this PR has all of the same changes.
2. Connected the leaderboard and profile to the backend so that all data is sourced from the database.
3. Leaderboard and profile change based on if a user is logged in.
4. Created base structure of the login page which enables users to log in and log out.
5. Created an error landing page for all undefined routes. 
6. Some backend changes regarding the data sent back when querying the /photo endpoint are included in this PR to allow the profile page to work.

Jest tests still need to be written for the Profile and PageNotFound components.

### CODE REVIEW

**Comment by JMStudiosJoe** [comment]
is this pr relying on #16 ?

**Comment by JMStudiosJoe** [comment]
do i merge 16 first or can this just be merged?

File: client/src/utils/capitalizeFirstLetter.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const capitalizeList = (list) => (
```
Comment (JMStudiosJoe) [inline client/src/utils/capitalizeFirstLetter.js:1]: is this for backend or for UI? for web UI this should be done in just CSS. However I know in React-native they dont but this aint react-native, yet.

Comment (JMStudiosJoe) [reply client/src/utils/capitalizeFirstLetter.js:1]: if this is just for web and just for the UI then lets keep it as CSS `text-decoration: 'capitolize'`  or something like that

File: client/src/utils/FetchUserData.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/* eslint-disable consistent-return */
+import Cookies from 'js-cookie';
+import loginAuth from './loginAuth';
+import URLpath from './URLpath';
+
+export async function fetchLogin(body) {
+  try {
+    const res = await fetch(URLpath('login'), {
```
Comment (JMStudiosJoe) [inline client/src/utils/FetchUserData.js:8]: I do like how this is one of the few places we are using fetch, great job on that. Id say lets take it a step further, make its own utils file where `fetch` is only used there. That way if need to use something else, we only have one place to change it. Does that make sense?

File: client/src/Components/Register/Login.jsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import Cookies from 'js-cookie';
+import { Link } from 'react-router-dom';
+import { fetchLogin, fetchLogOut } from '../../utils/fetchUserData';
+
```
Comment (JMStudiosJoe) [inline client/src/Components/Register/Login.jsx:5]: Components/Register/Login.jsx ? why is login within Register?

Comment (JMStudiosJoe) [reply client/src/Components/Register/Login.jsx:5]: if this is apart of previous pr my bad.

File: client/src/Components/Profile/ProfileStatistics.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+          <h4>{category[0]}</h4>
+        </div>
+        <p className="profile-stats-number">{category[1]}</p>
+      </div>
+      <hr />
+    </div>
+  );
+
+  // Capitalizes the first letter and sorts the list in descending order
+  useEffect(() => {
```
Comment (JMStudiosJoe) [inline client/src/Components/Profile/ProfileStatistics.jsx:40]: ah, yes please do with css class. CSS is much much faster than js and we can almost count all the steps js will go through. CSS is always better in web for how the UI should be displayed, and also is way faster in animtions last i checked.

**Review by JMStudiosJoe — CHANGES_REQUESTED** [review]
if any of my comments are from previous pr please respond to the comment saying so and dont fix.

**Comment by B-Tanonaka** [comment]
> is this pr relying on #16 ?

Yes these backend changes are in #16. Tim said he made the changes you requested in #16 and I pulled them into this branch.

**Comment by B-Tanonaka** [comment]
@JMStudiosJoe The requested changes have been pushed. Thank you for the feedback.

**Comment by JMStudiosJoe** [comment]
just merged pr #16 please pull the latest changes and update this pr and it should be a lot less changes and clean :) and then ready for merge. GREAT WORK!!!

**Comment by B-Tanonaka** [comment]
Just pulled the changes from main. This should be ready to merge now.

**Pull request merged**